3rdparty/mshadow/mshadow/tensor.h

```diff
@@ -69,15 +69,15 @@ struct Shape {
    * \param idx dimension index
    * \return the corresponding dimension size
    */
-  MSHADOW_XINLINE index_t &operator[](index_t idx) {
+  MSHADOW_XINLINE index_t &operator[](int idx) {
     return shape_[idx];
   }
   /*!
    * \brief get corresponding index
    * \param idx dimension index
    * \return the corresponding dimension size
    */
-  MSHADOW_XINLINE const index_t &operator[](index_t idx) const {
+  MSHADOW_XINLINE const index_t &operator[](int idx) const {
```

Comment:
There are other places in the same file that assumes idx have index_t type. we need to fix all of them.

Reply:
Done